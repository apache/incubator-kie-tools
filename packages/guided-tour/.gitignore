# dependencies
node_modules

# builds
build
dist
.rpt2_cache

# misc
**/.DS_Store
.env
.env.local
.env.development.local
.env.test.local
.env.production.local
*.log
coverage
target
out
.idea
.vscode
