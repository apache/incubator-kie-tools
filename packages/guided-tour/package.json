{
  "name": "@kie-tooling-core/guided-tour",
  "version": "0.0.0",
  "description": "Guided Tour component",
  "license": "Apache-2.0",
  "source": "src/index.tsx",
  "files": [
    "dist",
    "src"
  ],
  "scripts": {
    "lint": "yarn run run-script-if --bool \"$(build-env global.build.lint)\" --then \"yarn eslint ./src --ext .ts,.tsx\"",
    "test": "yarn run run-script-if --bool \"$(build-env global.build.test)\" --then \"jest --silent --verbose --passWithNoTests\"",
    "copy:css": "copyfiles -u 1 src/**/*.{sass,scss,css} dist/",
    "build:dev": "rimraf dist && yarn copy:css && tsc -p tsconfig.dev.json",
    "build:prod": "yarn lint && rimraf dist && yarn copy:css && tsc -p tsconfig.prod.json && yarn test"
  },
  "dependencies": {
    "@kie-tooling-core/envelope-bus": "0.0.0",
    "@kie-tooling-core/i18n": "0.0.0",
    "@kie-tooling-core/patternfly-base": "0.0.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@kogito-tooling/build-env": "0.0.0"
  },
  "presets": [
    [
      "@babel/env",
      {
        "targets": {
          "node": "current"
        }
      }
    ],
    "@babel/react"
  ]
}
