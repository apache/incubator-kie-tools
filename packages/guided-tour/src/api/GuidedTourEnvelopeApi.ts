import { Rect } from "../api";

export interface GuidedTourEnvelopeApi {
  receive_guidedTourElementPositionRequest(selector: string): Promise<Rect>;
}
