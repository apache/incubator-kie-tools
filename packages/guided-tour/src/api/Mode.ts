/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { UserInteraction } from ".";

export class BlockMode {
  // Each step shows something, and waits for some user interaction
  constructor(public userInteraction: UserInteraction, public allowedSelectors: string[]) {}
}

export class AutoMode {
  // Each step shows something, stays opened for some time, and it's auto-skipped
  constructor(public delay: number) {}
}

export class DemoMode {
  // Each step shows something, and waits for the "Next step" click
}

export class SubTutorialMode {
  // This steps englobes an amount of other sub-steps
  constructor(public label: string) {}
}

export type Mode = BlockMode | AutoMode | DemoMode | SubTutorialMode;
