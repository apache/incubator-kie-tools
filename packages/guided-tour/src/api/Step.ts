/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { DemoMode, Mode } from ".";

export class Step {
  constructor(
    public mode: Mode,
    public content?:
      | React.ReactNode
      | ((props: { dismiss?: () => void; nextStep?: () => void; prevStep?: () => void }) => React.ReactNode)
      | string,
    public selector?: string,
    public highlightEnabled?: boolean,
    public navigatorEnabled?: boolean,
    public position?: "right" | "bottom" | "center" | "left",
    public negativeReinforcementMessage?: string
  ) {}
}

export const NONE: Step = {
  mode: new DemoMode(),
};
