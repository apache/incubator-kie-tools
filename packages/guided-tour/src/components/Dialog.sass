/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 * 
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *        http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import "./Colors.sass"

.pf-c-modal-box.kgt-dialog
  position: fixed
  min-height: 160px
  width: max-content
  max-width: 600px
  padding: var(--pf-global--spacer--xl) var(--pf-global--spacer--xl) var(--pf-global--spacer--2xl)
  left: 50%
  top: 50%
  transition: all .5s ease-out

  & > .pf-c-button + *
    margin-right: 0

  &--enabled
    display: block

  &--disabled
    display: none

  .kgt-slide--with-accent
    margin: 0 32px
    text-align: center

    .pf-c-content--align-left
      text-align: left

    & > svg
      color: $kgt-accent-color
      height: 6em
      width: 6em
      margin: 8px 0

      & + .pf-c-title
        margin-top: 0
  
    & + .kgt-nav-controls
      display: none
