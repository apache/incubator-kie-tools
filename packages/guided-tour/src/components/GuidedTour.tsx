/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { useState } from "react";
import { Dialog, HighlightLayer } from ".";
import { UserInteraction, Tutorial, Rect } from "../api";
import { CurrentTutorialContext } from "../contexts";
import { I18nDictionariesProvider } from "@kie-tooling-core/i18n/dist/react-components";
import { GuidedTourI18nContext, guidedTourI18nDefaults, guidedTourI18nDictionaries } from "../i18n";

export const GuidedTour = () => {
  const [currentTutorial, setCurrentTutorial] = useState<Tutorial>();
  const [currentRefElementPosition, setCurrentRefElementPosition] = useState<Rect>();
  const [isNegativeReinforcementStateEnabled, setIsNegativeReinforcementStateEnabled] = useState(false);
  const [latestUserInteraction, setLatestUserInteraction] = useState<UserInteraction>();
  const [isHighlightLayerEnabled, setIsHighlightLayerEnabled] = useState(false);
  const [currentStep, setCurrentStep] = useState(0);
  const [completedStep, setCompletedStep] = useState(0);

  return (
    <I18nDictionariesProvider
      defaults={guidedTourI18nDefaults}
      dictionaries={guidedTourI18nDictionaries}
      initialLocale={navigator.language}
      ctx={GuidedTourI18nContext}
    >
      <CurrentTutorialContext.Provider
        value={{
          currentTutorial,
          currentStep,
          completedStep,
          currentRefElementPosition,
          isNegativeReinforcementStateEnabled,
          isHighlightLayerEnabled,
          latestUserInteraction,
          setCurrentStep,
          setCompletedStep,
          setCurrentTutorial,
          setCurrentRefElementPosition,
          setIsNegativeReinforcementStateEnabled,
          setIsHighlightLayerEnabled,
          setLatestUserInteraction,
        }}
      >
        <HighlightLayer />
        <Dialog isEnabled={false} tutorialLabel="" />
      </CurrentTutorialContext.Provider>
    </I18nDictionariesProvider>
  );
};
