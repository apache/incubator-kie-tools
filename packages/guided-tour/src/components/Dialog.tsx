/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { useState, useContext, useEffect, useMemo } from "react";

import {
  calculatePositionStyle,
  getCurrentStep,
  EmptyDialog,
  NegativeReinforcementDialog,
  StepDialog,
  usePositionListener,
  useSelectorHandler,
  useStartTutorialListener,
  useUserInteractionListener,
  useUserInteractions,
} from "./utils";
import { KogitoGuidedTour } from "..";
import { CurrentTutorialContext } from "../contexts";
import { DEFAULT_RECT, AutoMode, DemoMode } from "../api";

import "./Dialog.sass";
import { useGuidedTourI18n } from "../i18n";

export const Dialog = (props: { isEnabled: boolean; tutorialLabel: string }) => {
  // Local state
  const [isEnabled, setIsEnabled] = useState(props.isEnabled);
  const [currentTutorialLabel, setCurrentTutorialLabel] = useState(props.tutorialLabel);

  // Global state
  const {
    currentTutorial,
    currentStep,
    completedStep,
    currentRefElementPosition,
    isNegativeReinforcementStateEnabled,
    isHighlightLayerEnabled,
    setCurrentTutorial,
    setCurrentStep,
    setCompletedStep,
    setCurrentRefElementPosition,
    setLatestUserInteraction,
    setIsNegativeReinforcementStateEnabled,
    setIsHighlightLayerEnabled,
  } = useContext(CurrentTutorialContext);
  const guidedTour = KogitoGuidedTour.getInstance();
  const registeredTutorials = guidedTour.getRegisteredTutorials();

  // Aliases
  const dialogClass = "pf-c-modal-box kgt-dialog kgt-dialog" + (isEnabled ? "--enabled" : "--disabled");
  const dialogStep = useMemo(() => getCurrentStep(currentStep, currentTutorial), [currentStep, currentTutorial]);
  const dialogContent = dialogStep?.content ?? "";
  const dialogPosition = dialogStep?.position ?? "center";
  const dialogMode = dialogStep?.mode ?? new DemoMode();
  const dialogRefElement = isNegativeReinforcementStateEnabled ? DEFAULT_RECT : currentRefElementPosition;

  // Post processing
  const dialogStyle = calculatePositionStyle(dialogPosition, dialogRefElement);
  const emptyTemplate = useMemo(EmptyDialog(closeDialog), []);
  const regularTemplate = useMemo(StepDialog(dialogContent, closeDialog), [
    currentStep,
    currentTutorial,
    registeredTutorials,
    isNegativeReinforcementStateEnabled,
  ]);
  const negativeReinforcementTemplate = useMemo(NegativeReinforcementDialog(dialogStep, closeDialog), [
    currentStep,
    isHighlightLayerEnabled,
    isNegativeReinforcementStateEnabled,
  ]);

  useStartTutorialListener((tutorialLabel) => setCurrentTutorialLabel(tutorialLabel));
  usePositionListener((rect) => setCurrentRefElementPosition(rect));
  useUserInteractionListener((userInteraction) => setLatestUserInteraction(userInteraction));
  useUserInteractions();
  useSelectorHandler();

  function closeDialog() {
    setIsEnabled(false);
    guidedTour.teardown();
  }

  function getDialogTemplate() {
    if (isNegativeReinforcementStateEnabled && dialogStep) {
      return negativeReinforcementTemplate;
    } else if (dialogStep) {
      return regularTemplate;
    } else {
      return emptyTemplate;
    }
  }

  function handleAutoMode() {
    if ("delay" in dialogMode) {
      const autoMode = dialogMode as AutoMode;
      setTimeout(() => setCurrentStep(currentStep + 1), autoMode.delay);
    }
  }

  useEffect(() => {
    if (currentStep > completedStep) {
      setCompletedStep(currentStep);
    }
  }, [currentStep]);

  useEffect(() => {
    const newCurrentTutorial = registeredTutorials.find((tutorial) => tutorial.label === currentTutorialLabel);

    if (newCurrentTutorial) {
      setCurrentTutorial(newCurrentTutorial);
      setIsEnabled(true);
      setCurrentStep(0);
      setCompletedStep(0);
      setCurrentRefElementPosition(DEFAULT_RECT);
      setIsNegativeReinforcementStateEnabled(false);
      setIsHighlightLayerEnabled(false);
    }
  }, [registeredTutorials, currentTutorialLabel]);

  handleAutoMode();

  return (
    <div style={dialogStyle} className={dialogClass} data-ouia-component-id="dmn-guided-tour">
      {getDialogTemplate()}
    </div>
  );
};
