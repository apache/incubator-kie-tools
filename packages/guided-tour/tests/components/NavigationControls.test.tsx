/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { act } from "react-dom/test-utils";
import { render, renderedComponent, setupContainer, teardownContainer, triggerClick, useContextMock } from "../utils";

import { NavigationControls } from "@kie-tooling-core/guided-tour/dist/components";
import { KogitoGuidedTour } from "@kie-tooling-core/guided-tour/dist";
import { DemoMode } from "@kie-tooling-core/guided-tour/dist/api";

describe("NavigationControls", () => {
  beforeEach(setupContainer);
  afterEach(teardownContainer);

  it("works when users go to the next step", () => {
    const ctx = {
      currentStep: 2,
      setCurrentStep: (index: number) => {
        ctx.currentStep = index;
      },
      currentTutorial: { steps: [{}, {}, {}, {}] },
    };

    act(() => {
      useContextMock(ctx);
      render(<NavigationControls />);

      triggerClick("[data-kgt-next]");
    });

    expect(renderedComponent()).toMatchSnapshot();
    expect(ctx.currentStep).toBe(3);
  });

  it("works when users go to the previous step", () => {
    const ctx = {
      currentStep: 2,
      setCurrentStep: (index: number) => {
        ctx.currentStep = index;
      },
      currentTutorial: { steps: [{}, {}, {}, {}] },
    };

    act(() => {
      useContextMock(ctx);
      render(<NavigationControls />);

      triggerClick("[data-kgt-prev]");
    });

    expect(renderedComponent()).toMatchSnapshot();
    expect(ctx.currentStep).toBe(1);
  });

  it("renders with a tutorial and a step", () => {
    act(() => {
      useContextMock({
        currentStep: 2,
        currentTutorial: {
          steps: [{}, {}, {}],
        },
      });
      render(<NavigationControls />);
    });
    expect(renderedComponent()).toMatchSnapshot();
  });

  it("renders without a tutorial, but with a step", () => {
    act(() => {
      useContextMock({ currentStep: 2, currentTutorial: undefined });
      render(<NavigationControls />);
    });
    expect(renderedComponent()).toMatchSnapshot();
  });

  it("renders without a tutorial and without a step", () => {
    act(() => {
      useContextMock({ currentStep: undefined, currentTutorial: undefined });
      render(<NavigationControls />);
    });
    expect(renderedComponent()).toMatchSnapshot();
  });

  it("renders with buttons when navigator is enabled", () => {
    const ctx = {
      currentStep: 0,
      setCurrentStep: (index: number) => {
        ctx.currentStep = index;
      },
      currentTutorial: {
        label: "Tutorial",
        steps: [
          {
            mode: new DemoMode(),
            navigatorEnabled: true,
          },
        ],
      },
    };

    act(() => {
      KogitoGuidedTour.getInstance().registerTutorial(ctx.currentTutorial);
      useContextMock(ctx);
      render(<NavigationControls />);
    });

    expect(renderedComponent()).toMatchSnapshot();
  });
});
