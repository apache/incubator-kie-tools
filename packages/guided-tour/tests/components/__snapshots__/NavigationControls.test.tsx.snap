// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationControls renders with a tutorial and a step 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-5"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    >
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet kgt-nav-controls__bullet--current"
      >
          
      </div>
    </div>
    <button
      aria-disabled="true"
      class="pf-c-button pf-m-link pf-m-disabled"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-6"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      disabled=""
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`NavigationControls renders with buttons when navigator is enabled 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="true"
      class="pf-c-button pf-m-link pf-m-disabled"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-11"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    >
      <div
        class="kgt-nav-controls__bullet kgt-nav-controls__bullet--current"
      >
          
      </div>
    </div>
    <button
      aria-disabled="true"
      class="pf-c-button pf-m-link pf-m-disabled"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-12"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`NavigationControls renders without a tutorial and without a step 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-9"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    />
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-10"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`NavigationControls renders without a tutorial, but with a step 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-7"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    />
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-8"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`NavigationControls works when users go to the next step 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    >
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet kgt-nav-controls__bullet--current"
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-2"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`NavigationControls works when users go to the previous step 1`] = `
<div>
  <div
    class="kgt-nav-controls"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-prev="true"
      data-ouia-component-id="OUIA-Generated-Button-link-3"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <div
      class="kgt-nav-controls__bullets"
    >
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet kgt-nav-controls__bullet--current"
      >
          
      </div>
      <div
        class="kgt-nav-controls__bullet "
      >
          
      </div>
    </div>
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-link"
      data-kgt-next="true"
      data-ouia-component-id="OUIA-Generated-Button-link-4"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      hidden=""
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>
</div>
`;
